### ISSUE TITLE
Ability to start/pause/stop

### ISSUE THREAD

**Issue by binarykitchen** [livejs/read-audio#7]
This through the stream object somehow.

For inspiration, check out this code of mine
https://github.com/binarykitchen/videomail-client/blob/master/src/util/audioRecorder.js

My aim is to move this to your package whenever your package comes with all the functionality I need.

**Comment by binarykitchen** [comment]
@ahdinosaur ping?

**Comment by ahdinosaur** [comment]
too busy at the moment with other work, sorry. happy to merge any pull requests and / or add you as a repo collaborator.

**Comment by binarykitchen** [comment]
i can wait and just were wondering about an update

**Comment by binarykitchen** [comment]
@ahdinosaur ping?

**Comment by ahdinosaur** [comment]
hey @binarykitchen!

so i just pushed a re-write of this module as a [source pull stream](https://github.com/dominictarr/pull-stream) (see [`pull-stream-examples`](https://github.com/dominictarr/pull-stream-examples) and [yosh's blob post](https://medium.com/@yoshuawuyts/streams-in-node-ab9f13e15d5) for why i switched, tl;dr pull streams are simple functions).

this means you should be able to use [`pull-pause`](https://github.com/dominictarr/pull-pause) to start / pause / stop the stream, which i'd prefer as it'd keep this module specific to capturing the audio.

**Comment by binarykitchen** [comment]
nice, good work mate. i ll check this out soon! is this documented in the readme??

and what about the other open tickets? reason i ask is because i need them all for https://github.com/binarykitchen/videomail-client (not some of them but all of them due to unfortunately-high-coupled code)

**Comment by ahdinosaur** [comment]
> is this documented in the readme??

yep as best i could, but please lemme know what's missing. :smile: 

> and what about the other open tickets? reason i ask is because i need them all for https://github.com/binarykitchen/videomail-client (not some of them but all of them due to unfortunately-high-coupled code)

yep, aware of what you need and would like to help you out, but limited by my own capacity and that i'm not currently working on any projects that need this module. still tried [doing type checking](https://github.com/livejs/read-audio/tree/type-checking) to solve #6 and #9, have some bugs to sort out.

anyways, thanks for your interest in this module and http://videomail.io seems sweet!

**Comment by binarykitchen** [comment]
@ahdinosaur thanks heaps mate

ok, had a look at your readme example. for pause/resume, were you thinking of something like this?

``` js
var readAudio = require('read-audio')
var pull = require('pull-stream')
var pause = require('pull-pause')()

pull(
  readAudio(),
  pull.map(function (arr, enc, cb) {
    // do something
  }),
  pause,
  pull.drain(function (str) {
    process.stdout.write(str)
  })
)

pause.pause() //stop reading.

pause.resume() //resume reading.
```

for the other tickets, okay, i understand, take your time. just hope somewhen this year :)

**Comment by ahdinosaur** [comment]
yeah, that should work, probably should add that example to the README. :smile:
